config/cluster-install/rbac/numaflow-server/numaflow-server-cluster-role.yaml

```diff
@@ -37,13 +37,15 @@ rules:
       - pods/log
       - pods/exec
       - configmaps
+      - secrets
```

Comment:
For this, use a separated `Role` and `RoleBinding` - the UX server does not need to have secret access to cluster scope.